### DISCUSSION TITLE
how to batch convert  multiple jpeg images to tiff images and compress using libvips in command prompt....

### DISCUSSION THREAD

**Discussion by shameerasl** [libvips/libvips#2738] [Q&A] [answered]
hai... I am new to libvips.If I am blundering please forgive.I would like to batch convert a lot of files from jpeg to tif. and also adding compression and tyle pyramidal. any reference will be great help.. thank you in advance.

**Comment by jcupitt** [comment, marked as answer]
Hi @shameerasl,

You need to write a small script. On linux, something like:

```bash
for name in *.jpg; do
    vips copy $name $name.tif[compression=jpeg,tile,pyramid]
done
```

**Reply by shameerasl** [reply]
thanks for your help. what about it in windows os..

**Reply by jcupitt** [reply]
You'll need to write a small BAT file, or perhaps use powershell.

You could also use something like Python I suppose.

**Reply by shameerasl** [reply]
thank you @jcupitt .i wrotea bat file. and it worked well. 
I have another issue like i want to megre tiff images in a folder to one pyramidal tiff image. how to do this in windows cmd using libvip

**Reply by shameerasl** [reply]
i used array join . it works fine in the following code
vips arrayjoin "a.tif b.tif c.tif d.tif" out.tif --across 2
but after using this
 vips arrayjoin "a.tif b.tif c.tif d.tif" out.tif[bigtiff,compression=jpg,tile,pyramid] --across 2
it shows too many arguments.  any reference will be great help.. thank you in advance.

**Reply by jcupitt** [reply]
It should work. It sounds like you have a quoting problem in your BAT file. I think you'll need to ask a BAT file expert.

**Reply by powerlicht** [reply]
Too bad the topic starter didn't provide more details. I tried to recreate the problem from the latter with a version from 2021, but could not reproduce the error mentioned. But I got to know a new function **_arrayjoin_**
